mutator-common/src/tmp_package_dir.rs

```diff
@@ -0,0 +1,68 @@
+//! A path setup container for packages under test.
+// Copyright © Eiger
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use fs_extra::dir::CopyOptions;
+use log::{info, trace};
+use move_package::source_package::layout::SourcePackageLayout;
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+/// The path in temp dir where the original Move package is cloned.
+const ORIGINAL_PACKAGE_PATH: &str = "original_package";
+
+/// Returns the output directory and a recreated package path.
+pub fn setup_outdir_and_package_path<P: AsRef<Path>>(
+    package_path: P,
+) -> Result<(PathBuf, PathBuf)> {
+    // Check if the package is correctly structured.
+    let package_path = SourcePackageLayout::try_find_root(&package_path.as_ref().canonicalize()?)?;
+    info!("Found package path: {package_path:?}");
+
+    let outdir = tempfile::tempdir()?.into_path();
+    let new_package_path = outdir.join(ORIGINAL_PACKAGE_PATH);
+    fs::create_dir_all(&new_package_path)?;
+
+    let options = CopyOptions::new().content_only(true);
+    fs_extra::dir::copy(&package_path, &new_package_path, &options)?;
+
+    move_mutator::compiler::rewrite_manifest_for_mutant(&package_path, &new_package_path)?;
+
+    let remove_dir = |dir_name| {
+        let dir = new_package_path.join(dir_name);
+        if fs::remove_dir_all(&dir).is_ok() {
+            trace!("removing {}", dir.display());
+        }
+    };
+    let remove_file = |dir_name| {
+        let dir = new_package_path.join(dir_name);
+        if fs::remove_file(&dir).is_ok() {
+            trace!("removing {}", dir.display());
+        }
+    };
+
+    // Since the tool will copy the original package often, remove unnecessary files.
+    remove_dir("build");
+    remove_dir("doc");
+    remove_file(".trace");
+    remove_file("output.bpl");
```

Comment:
Something like that could be shorter alternative

```suggestion
    let remove_item = |item_name: &str, is_dir: bool| {
        let item_path = new_package_path.join(item_name);
        let result = if is_dir {
            fs::remove_dir_all(&item_path)
        } else {
            fs::remove_file(&item_path)
        };
        if result.is_ok() {
            trace!("Removed {}", item_path.display());
        }
    };

    ["build", "doc"].iter().for_each(|&dir| remove_item(dir, true));
    [".trace", "output.bpl"].iter().for_each(|&file| remove_item(file, false));
```

Reply:
Done